From bc0c714e24816ba09cead78d9fd2474024b0168e Mon Sep 17 00:00:00 2001
From: kszaq <737984+kszaq@users.noreply.github.com>
Date: Thu, 29 Dec 2022 20:19:58 +0100
Subject: [PATCH] AMLCodec: always try to dequeue buffer in GetPicture

Return VC_BUFFER only if codec can't dequeue data (get new picture) and buffer is not full.
---
 xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index ffa268eb33..39e7ff4612 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -2531,10 +2531,6 @@ CDVDVideoCodec::VCReturn CAMLCodec::GetPicture(VideoPicture *pVideoPicture)
   if (!m_opened)
     return CDVDVideoCodec::VC_ERROR;
 
-  float timesize(GetTimeSize());
-  if(!m_drain && timesize < 0.2f)
-    return CDVDVideoCodec::VC_BUFFER;
-
   if (DequeueBuffer() == 0)
   {
     pVideoPicture->iFlags = 0;
@@ -2560,7 +2556,7 @@ CDVDVideoCodec::VCReturn CAMLCodec::GetPicture(VideoPicture *pVideoPicture)
   }
   else if (m_drain)
     return CDVDVideoCodec::VC_EOF;
-  else if (timesize < 1.0f)
+  else if (GetTimeSize() < 1.0f)
     return CDVDVideoCodec::VC_BUFFER;
 
   return CDVDVideoCodec::VC_NONE;
